cpp/src/parquet/stream_reader_test.cc

```diff
@@ -912,5 +913,39 @@ TEST_F(TestReadingDataFiles, Int64Decimal) {
   EXPECT_EQ(i, 25);
 }
 
+TEST_F(TestReadingDataFiles, FLBADecimal) {
+  PARQUET_ASSIGN_OR_THROW(auto infile, ::arrow::io::ReadableFile::Open(
+                                           GetDataFile("fixed_length_decimal.parquet")));
+
+  auto file_reader = ParquetFileReader::Open(infile);
+  auto reader = StreamReader{std::move(file_reader)};
+
+  ::arrow::Decimal128 x;
+  int i;
```

Comment:
`int i = 0;` to avoid reader already eof, and `i` is undefined?

Reply:
I just follow the same pattern of previous cases. These files are well-defined so i will always be updated.